### PR TITLE
use celestiaorg instead of lazyledger

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

replaces lazyledger-core with celestia-core and updates to the latest commit of celestia-core

blocked by: [#522](https://github.com/celestiaorg/celestia-core/pull/522)
closes: #26

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,53 +7,52 @@ require (
...
+	google.golang.org/protobuf v1.26.0
 	gopkg.in/yaml.v2 v2.4.0
 )
 
-replace github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.2-alpha.regen.4
+replace (
+	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.2-alpha.regen.4
+	github.com/golang/mock => github.com/golang/mock v1.4.4
+	google.golang.org/grpc => google.golang.org/grpc v1.33.2
+)
```
Comment (evan-forbes) [inline go.mod:58]: had to use older versions of `golang/mock` and `grpc` to get tests to pass.

upstream sdk still uses v1.33.2 of grpc due to the same bugs encountered here [ref](https://github.com/cosmos/cosmos-sdk/issues/8426) Should be fixed in a future [update](https://github.com/cosmos/cosmos-sdk/issues/8392#issuecomment-795771954). 

As for `golang/mock`, I think an updated version of the sdk will accommodate the updated version, so we can remove that replace when we update.

File: go.mod [hunk 2]
```diff
@@ -7,53 +7,52 @@ require (
 	github.com/DataDog/zstd v1.4.5 // indirect
 	github.com/armon/go-metrics v0.3.6
 	github.com/bgentry/speakeasy v0.1.0
-	github.com/btcsuite/btcd v0.21.0-beta
-	github.com/btcsuite/btcutil v1.0.2
+	github.com/btcsuite/btcd v0.22.0-beta
+	github.com/btcsuite/btcutil v1.0.3-0.20201208143702-a53e38424cce
+	github.com/celestiaorg/celestia-core v0.0.1-mvp-das-lightclient.0.20210830200007-52e4383a7266
```
Comment (evan-forbes) [inline go.mod:12]: I'll also update this to use the master branch as soon as we merge [#522](https://github.com/celestiaorg/celestia-core/pull/522)

**Comment by evan-forbes** [comment]
go-lint is expected to fail due to using standard lib instead of tendermint one (see [ref](https://github.com/celestiaorg/cosmos-sdk/pull/13#issuecomment-764022556).

File: third_party/proto/tendermint/abci/types.proto [hunk 3]
```diff
@@ -370,7 +370,7 @@ message Evidence {
   ];
   // Total voting power of the validator set in case the ABCI application does
   // not store historical validators.
-  // https://github.com/lazyledger/lazyledger-core/issues/4581
+  // https://github.com/celestiaorg/celestia-core/issues/4581
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply third_party/proto/tendermint/abci/types.proto]: nice catch!! resolved in https://github.com/celestiaorg/cosmos-sdk/pull/28/commits/21f0b785b153d660ee1fb64c0e4785108fec81fc

File: x/evidence/types/query.pb.go [hunk 4]
```diff
@@ -6,12 +6,12 @@ package types
 import (
 	context "context"
 	fmt "fmt"
+	github_com_lazyledger_lazyledger_core_libs_bytes "github.com/celestiaorg/celestia-core/libs/bytes"
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply x/evidence/types/query.pb.go]: whoops, forgot to regenerate the proto files :eyes: 

thanks for finding this! resolved in https://github.com/celestiaorg/cosmos-sdk/pull/28/commits/45bbca67e7f2e2cddcbe7dbd33ef5048206195b2

File: simapp/simd/cmd/testnet.go [hunk 5]
```diff
@@ -116,7 +116,7 @@ func InitTestnet(
 ) error {
 
 	if chainID == "" {
-		chainID = "chain-" + tmrand.NewRand().Str(6)
+		chainID = "chain-" + tmrand.Str(6)
```
[the remark this answers is not in the document]
Comment (evan-forbes) [reply simapp/simd/cmd/testnet.go:119]: good question, check out https://github.com/celestiaorg/celestia-core/pull/284

**Comment by evan-forbes** [comment]
also, the liveness test is passing for me locally, so I don't think it's worth debugging the CI atm. If it still fails after we update to the latest version of the sdk, then we can debug it then.

**Comment by evan-forbes** [comment]
I think we can just update to the latest version of the sdk and see if they updated it. leaving a ref #27 to remind myself

**Review by Bidon15 — APPROVED** [review]

**Comment by Bidon15** [comment]
Let's hope that it will self-cure after the update 🤞 as we will be doomed to do this manual labour omt 😓

**Review by Bidon15 — APPROVED** [review]
👍

**Review by Bidon15 — APPROVED** [review]

**Review by tzdybal — APPROVED** [review]
LGTM

**Pull request merged**